diff --git a/rotators/ars/ars.c b/rotators/ars/ars.c
index 1b8e4b0..a8a8a0f 100644
--- a/rotators/ars/ars.c
+++ b/rotators/ars/ars.c
@@ -215,7 +215,11 @@ ars_close(ROT *rot)
 {
     struct ars_priv_data *priv = (struct ars_priv_data *)ROTSTATE(rot)->priv;
 
+#if defined(__ANDROID__)
+    (void)priv;
+#else
     pthread_cancel(priv->thread);
+#endif
 
     /* leave it in safe state */
     ars_stop(rot);
